### PR TITLE
Sta 405 room persistance

### PR DESCRIPTION
**Pull request by MisoMurin**
- database class, first dao and entity annotations added
- created database
- response wrappers modified - getter methods for api data

### CODE REVIEW

File: sdk/src/main/java/com/sygic/travel/sdk/StSDK.kt [hunk 1]
```diff
@@ -222,8 +224,10 @@ class StSDK internal constructor() {
     companion object {
         private val ST_SDK_NAME_AND_VERSION = "sygic-travel-sdk-android/sdk-version"
         private val ANDROID_VERSION = "Android/" + Build.VERSION.RELEASE
+	    private val DATABASE_NAME = "st-sdk-db"
```
Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/StSDK.kt]: indentation

File: sdk/src/main/java/com/sygic/travel/sdk/StSDK.kt [hunk 2]
```diff
@@ -243,6 +247,13 @@ class StSDK internal constructor() {
         fun initialize(xApiKey: String, context: Context) {
             StApiGenerator.headersInterceptor.setApiKey(xApiKey)
             StApiGenerator.headersInterceptor.setUserAgent(createUserAgent(context))
+
+            stDb = Room.databaseBuilder(
+                    context,
+                    StDb::class.java,
+		            DATABASE_NAME
```
Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/StSDK.kt]: indentation.

File: sdk/src/main/java/com/sygic/travel/sdk/api/model/place/ApiPlaceDetail.kt [hunk 3]
```diff
@@ -9,9 +10,11 @@ import com.sygic.travel.sdk.model.place.Place
 import com.sygic.travel.sdk.model.place.Reference
 import com.sygic.travel.sdk.model.place.Tag
 
+@Entity(tableName = "place")
```
Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/api/model/place/ApiPlaceDetail.kt]: should be store only favorites?

Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/api/model/place/ApiPlaceDetail.kt]: also, why we are storing api response and not our model entity?

File: sdk/src/main/java/com/sygic/travel/sdk/db/Converters.kt [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    @TypeConverter
+    fun stringToList(value: String): List<String> {
+        return value.split((",").toRegex(), 0)
+    }
+
+    @TypeConverter
+    fun listToString(values: List<String>): String {
+        return values.joinToString(",")
+    }
+}
```
Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/db/Converters.kt]: empty line on the end of the file

File: sdk/src/main/java/com/sygic/travel/sdk/db/StDb.kt [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import com.sygic.travel.sdk.db.dao.PlaceDetailDao
+
+@Database(
+		entities = arrayOf(ApiPlaceDetail::class),
+		version = 1
+)
+@TypeConverters(Converters::class)
+internal abstract class StDb : RoomDatabase() {
+	abstract fun placeDetailDao(): PlaceDetailDao
+}
```
Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/db/StDb.kt]: the same...

File: sdk/src/main/java/com/sygic/travel/sdk/db/dao/PlaceDetailDao.kt [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+@Dao
+internal interface PlaceDetailDao {
+    @Query("SELECT * FROM place")
+    fun getAll(): List<ApiPlaceDetail>
+
+    @Query("SELECT * FROM place WHERE id IN (:arg0)")
+    fun loadAllByIds(ids: List<Int>): List<ApiPlaceDetail>
+
+    @Query("SELECT * FROM place WHERE id=:arg0")
+    fun findById(id: String): ApiPlaceDetail
```
Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/db/dao/PlaceDetailDao.kt]: I suggest

- find for collections (lists)
- get for entities

File: sdk/src/main/java/com/sygic/travel/sdk/db/dao/PlaceDetailDao.kt [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+@Dao
+internal interface PlaceDetailDao {
+    @Query("SELECT * FROM place")
+    fun getAll(): List<ApiPlaceDetail>
+
+    @Query("SELECT * FROM place WHERE id IN (:arg0)")
+    fun loadAllByIds(ids: List<Int>): List<ApiPlaceDetail>
+
+    @Query("SELECT * FROM place WHERE id=:arg0")
```
Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/db/dao/PlaceDetailDao.kt]: coding style id = :arg0
also, isn't possible to name args after the function arg's name?

File: sdk/src/main/java/com/sygic/travel/sdk/db/dao/PlaceDetailDao.kt [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+    @Insert
+    fun insert(vararg places: ApiPlaceDetail)
+
+    @Insert
+    fun insert(places: List<ApiPlaceDetail>)
+
+    @Delete
+    fun delete(place: ApiPlaceDetail)
+}
```
Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/db/dao/PlaceDetailDao.kt]: empty line

File: sdk/src/main/java/com/sygic/travel/sdk/db/dao/PlaceDetailDao.kt [hunk 9]
```diff
@@ -17,10 +18,10 @@ internal interface PlaceDetailDao {
     @Query("SELECT * FROM place WHERE id=:arg0")
     fun findById(id: String): ApiPlaceDetail
 
-    @Insert
+    @Insert(onConflict = REPLACE)
```
Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/db/dao/PlaceDetailDao.kt]: it this a good default behavior? is it possible to somehow disable this?

File: sdk/src/main/java/com/sygic/travel/sdk/api/model/place/ApiTour.kt [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+		tour.rating = rating
+		tour.reviewCount = reviewCount
+		tour.photoUrl = photoUrl
+		tour.price = price
+		tour.originalPrice = originalPrice
+		tour.duration = duration
+
+		return tour
+	}
+}
```
Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/api/model/place/ApiTour.kt]: empty line

File: sdk/src/main/java/com/sygic/travel/sdk/db/dao/PlaceDetailDao.kt (hunk 6 above)
Comment (MisoMurin) [inline sdk/src/main/java/com/sygic/travel/sdk/db/dao/PlaceDetailDao.kt]: ✔️

File: sdk/src/main/java/com/sygic/travel/sdk/db/dao/PlaceDao.kt [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+import android.arch.persistence.room.Dao
+import android.arch.persistence.room.Delete
+import android.arch.persistence.room.Insert
+import android.arch.persistence.room.Query
+import com.sygic.travel.sdk.db.model.place.DbPlaceDetail
+
+@Dao
+internal interface PlaceDao {
+    @Query("SELECT * FROM place")
+    fun getAll(): List<DbPlaceDetail>
```
Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/db/dao/PlaceDao.kt]: findAll()

File: sdk/src/main/java/com/sygic/travel/sdk/db/dao/PlaceDao.kt [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+
+    @Insert
+    fun insert(vararg places: DbPlaceDetail)
+
+    @Insert
+    fun insert(places: List<DbPlaceDetail>)
+
+    @Delete
+    fun delete(place: DbPlaceDetail)
+}
```
Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/db/dao/PlaceDao.kt]: empty line

**Pull request merged**